routers/repo/issue.go

```diff
@@ -1256,3 +1258,51 @@ func DeleteMilestone(ctx *context.Context) {
 		"redirect": ctx.Repo.RepoLink + "/milestones",
 	})
 }
+
+// ChangeIssueReaction create a reaction for issue
+func ChangeIssueReaction(ctx *context.Context, form auth.ReactionForm) {
+	issue := GetActionIssue(ctx)
+	if ctx.Written() {
+		return
+	}
+
+	if ctx.HasError() {
```

Comment:
What does this `if` block accomplish?

Reply:
I think yes but I will verify. It's to check if there is form validation error